### PR TITLE
[WIP] Added expandable text to long delay messages

### PR DESCRIPTION
**Pull request by anjandev**
As per #422 . Please test this. Transportr does not support where I live and I could not completely test this patch.

### CODE REVIEW

**Comment by anjandev** [comment]
How do I sign the cla?

**Comment by grote** [comment]
You can just state here that you accept the agreement. That should be fine.

File: app/src/main/java/de/grobox/transportr/departures/ExpandableTextView.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		super(context, attrs);
+
+		TypedArray typedArray = context.obtainStyledAttributes(attrs, R.styleable.ExpandableTextView);
+		this.trimLength = typedArray.getInt(R.styleable.ExpandableTextView_trimLength, DEFAULT_TRIM_LENGTH);
+		typedArray.recycle();
+
+		setOnClickListener(new OnClickListener() {
+			@Override
+			public void onClick(View v) {
+				trim = !trim;
```
Comment (grote) [inline app/src/main/java/de/grobox/transportr/departures/ExpandableTextView.java]: How does this behave when rotating the phone? I couldn't see any code that stores the value of `trim`.

File: app/src/main/res/values/attrs.xml [hunk 2]
```diff
@@ -10,4 +10,8 @@
         <attr name="hint" format="string"/>
     </declare-styleable>
 
+    <declare-styleable name="ExpandableTextView">
+        <attr name="trimLength" format="integer"/>
+    </declare-styleable>
+
```
Comment (grote) [inline app/src/main/res/values/attrs.xml]: This seems to be unused.

File: app/src/main/java/de/grobox/transportr/departures/ExpandableTextView.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.grobox.transportr.departures;
+
+/**
+ * Created by anjan on 06/05/18.
+ * From: https://stackoverflow.com/questions/15627530/android-expandable-textview-with-animation
```
[the remark this answers is not in the document]
Comment (anjandev) [reply app/src/main/java/de/grobox/transportr/departures/ExpandableTextView.java]: https://stackexchange.com/legal#licensing
Under "Subscriber content" in Section 6, it states:
"You agree that any and all content, including without limitation any and all text, graphics, logos, tools, photographs, images, illustrations, software or source code, audio and video, animations, and product feedback (collectively, "Content") that you provide to the public Network (collectively, "Subscriber Content"), is perpetually and irrevocably licensed to Stack Overflow on a worldwide, royalty-free, non-exclusive basis pursuant to Creative Commons licensing terms (CC-BY-SA)."

Further clarifying and linking to the exact CC-BY-SA Creative Commons license in the same section:
"The CC-BY-SA Creative Commons license terms are explained in further detail by [Creative Commons](https://creativecommons.org/licenses/by-sa/4.0/)"

On October 8, 2015, Creative Commons concluded that the CC BY-SA 4.0 is one-way compatible with the GPLv3. See:
https://creativecommons.org/share-your-work/licensing-considerations/compatible-licenses/

[the remark this answers is not in the document]
Comment (anjandev) [reply app/src/main/java/de/grobox/transportr/departures/ExpandableTextView.java]: I'll try to read through these documents but I dont have a lot of experience with licensing software so I may need assistance.

[the remark this answers is not in the document]
Comment (anjandev) [reply app/src/main/java/de/grobox/transportr/departures/ExpandableTextView.java]: The stackexchange licensing page links to BY-SA 4.0 not cc by-sa 3.0:
https://stackexchange.com/legal

and as stated in my previous comments. I can rewrite the code if need be.

**Comment by anjandev** [comment]
I accept the cla

File: app/src/main/java/de/grobox/transportr/departures/ExpandableTextView.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (anjandev) [reply app/src/main/java/de/grobox/transportr/departures/ExpandableTextView.java]: Ya, that sounds like a much better solution. Ill take a look at both and try to implement one of them. I feel that this pull request can be closed.

**Comment by anjandev** [comment]
I found a much better library to do this with. I will be cleaning up my code tomorrow morning and pushing it to be merged.

**Pull request closed without merging**